jwst/extract_2d/extract_2d.py

```diff
@@ -17,54 +18,62 @@
-        log.info('Subarray x-extents are: %s %s', xlo, xhi)
-        log.info('Subarray y-extents are: %s %s', ylo, yhi)
+            log.info('Name of subarray extracted: %s', slit.name)
+            log.info('Subarray x-extents are: %s %s', xlo, xhi)
+            log.info('Subarray y-extents are: %s %s', ylo, yhi)
 
-        ext_data = input_model.data[ylo: yhi, xlo: xhi].copy()
-        ext_err = input_model.err[ylo: yhi, xlo: xhi].copy()
-        ext_dq = input_model.dq[ylo: yhi, xlo: xhi].copy()
-        new_model = datamodels.ImageModel(data=ext_data, err=ext_err, dq=ext_dq)
-        shape = ext_data.shape
-        domain = [{'lower': 0, 'upper': shape[1], 'includes_lower': True, 'includes_upper': False},
-                  {'lower':0, 'upper': shape[0], 'includes_lower': True, 'includes_upper': False}]
-        slit_wcs.domain = domain
-        new_model.meta.wcs = slit_wcs
-        output_model.slits.append(new_model)
-        # set x/ystart values relative to the image (screen) frame.
-        # The overall subarray offset is recorded in model.meta.subarray.
-        nslit = len(output_model.slits) - 1
-        output_model.slits[nslit].name = str(slit.name)
-        #output_model.slits[nslit].source_id = slit.source_id
-        output_model.slits[nslit].xstart = xlo + 1
-        output_model.slits[nslit].xsize = xhi - xlo
-        output_model.slits[nslit].ystart = ylo + 1
-        output_model.slits[nslit].ysize = yhi - ylo
+            ext_data = input_model.data[ylo: yhi, xlo: xhi].copy()
+            ext_err = input_model.err[ylo: yhi, xlo: xhi].copy()
+            ext_dq = input_model.dq[ylo: yhi, xlo: xhi].copy()
+            new_model = datamodels.ImageModel(data=ext_data, err=ext_err, dq=ext_dq)
+            shape = ext_data.shape
+            domain = [{'lower': 0, 'upper': shape[1], 'includes_lower': True, 'includes_upper': False},
+                      {'lower':0, 'upper': shape[0], 'includes_lower': True, 'includes_upper': False}]
+            slit_wcs.domain = domain
+            new_model.meta.wcs = slit_wcs
+            output_model.slits.append(new_model)
+            # set x/ystart values relative to the image (screen) frame.
+            # The overall subarray offset is recorded in model.meta.subarray.
+            nslit = len(output_model.slits) - 1
+            output_model.slits[nslit].name = str(slit.name)
+            output_model.slits[nslit].xstart = xlo + 1
+            output_model.slits[nslit].xsize = xhi - xlo
+            output_model.slits[nslit].ystart = ylo + 1
+            output_model.slits[nslit].ysize = yhi - ylo
+            if exp_type.lower() == 'nrs_msaspec':
+                output_model.slits[nslit].source_id = int(slit.source_id)
+                output_model.slits[nslit].source_name = slit.source_name
+                output_model.slits[nslit].source_alias = slit.source_alias
+                output_model.slits[nslit].catalog_id  = int(slit.catalog_id)
```

Comment:
We need to check with Lisa Sherbert to confirm the data types for a few of these entries, like source_id and catalog_id. The correct data type may not be the same as the data type in the few example MSA config files we got many months ago. We used to think the catalog_id would be a simple number, but I believe it's actually going to be a string giving the name of the catalog from which the sources come.